check_addon.py

```diff
@@ -334,6 +339,52 @@ def _check_file_whitelist(error_counter, file_index, addon_path):
     return error_counter
 
 
+def get_repo_addons(branch):
+    branch_url = ROOT_URL.format(branch=branch)
+    content = requests.get(branch_url).content
+    tree = ET.fromstring(content)
+
+    return {
+        a.get("id"): (a.get("version"), bool(a.find("extension/broken")))
+        for a in tree.findall("addon")
+    }
+
+
+def get_addon_name(xml_path):
+    tree = ET.parse(xml_path).getroot()
+    return (tree.get("id"), tree.get("version"))
+
+
+def _check_for_existing_addon(error_counter, addon_xml):
+    addon_xml = os.path.join(addon_xml, "addon.xml")
+
+    branches = ['gotham', 'helix', 'isengard', 'jarvis', 'krypton', 'leia']
+
+    current_branch = os.environ.get("TRAVIS_BRANCH")
+
+    addon_name, addon_version = get_addon_name(addon_xml)
+
+    for branch in branches:
+
+        if branch >= current_branch:
+            break
+
+        else:
+            repo_addons = get_repo_addons(branch)
+
+            if addon_name not in repo_addons:
+                error_counter = _logWarning(error_counter, "This is a new addon")
```

Comment:
With the current for loop over all branches, this might be wrong and/or logged multiple times for the same addon.

Reply:
Yeah, right. Should I just remove this check then? If it doesn't already exist in any branch then the addon is new.